### PR TITLE
Add system groovy

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
Added support for system groovy script in the Jenkins xml data

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java [hunk 1]
```diff
@@ -113,21 +125,51 @@ public JenkinsXmlData addBashScript(String script) {
...
         JSONObject newGroovyCommand = BuildGroovyObject(script);
 
-        builders.getJSONArray(hudsonGroovyKey)
+        builders.getJSONArray(hudsonPluginsGroovyGroovyKey)
                 .put(newGroovyCommand);
         return this;
     }
 
+    /**
+     * This function adds system groovy script to the XML data.
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: Write in this description that onyl one groovy script may be added,

Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: Done.

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java [hunk 2]
```diff
@@ -37,6 +40,15 @@
     private static final String XML_VERSION = "<?xml version='1.1' encoding='UTF-8'?>";
     private static final String HUDSON_PARAMETERS_DEFINITION_KEY = "hudson.model.ParametersDefinitionProperty";
     private static final String PARAMETER_DEFINITION_KEY = "parameterDefinitions";
+    private static final String GROOVY_SCRIPT_SCRIPT_PLUGIN_VERSION = "2.1";
+    private static final String GROOVY_SCRIPT_SECURITY_PLUGIN_VERSION = "1.51";
+
+    private String hudsonPluginsGroovyGroovyKey = "hudson.plugins.groovy.Groovy plugin='groovy@%s'";
```
Comment (tobiasake) [inline src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: Why groovy two times in the name?

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java [hunk 3]
```diff
@@ -113,21 +125,52 @@ public JenkinsXmlData addBashScript(String script) {
      * @return this JenkinsXmlData
      */
     public JenkinsXmlData addGrovyScript(String script) {
-        String hudsonGroovyKey = "hudson.plugins.groovy.Groovy plugin='groovy@2.1'";
-        boolean hasKeyGroovy = builders.has(hudsonGroovyKey);
+        hudsonPluginsGroovyGroovyKey = String.format(hudsonPluginsGroovyGroovyKey, GROOVY_SCRIPT_SCRIPT_PLUGIN_VERSION);
```
Comment (tobiasake) [inline src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: Why Script two times in the name?

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java [hunk 4]
```diff
@@ -113,21 +125,52 @@ public JenkinsXmlData addBashScript(String script) {
...
+     * Only one system groovy script may be added to a single jenkins job at the moment.
+     *
+     * @param script
+     * @param sandbox
+     * @return this JenkinsXmlData
+     * @throws Exception
+     */
+    public JenkinsXmlData addSystemGrovyScript(String script, boolean sandbox) throws Exception {
+        hudsonPluginGroovySystemGroovyKey = String.format(hudsonPluginGroovySystemGroovyKey,
+                GROOVY_SCRIPT_SCRIPT_PLUGIN_VERSION);
```
Comment (tobiasake) [inline src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: same comment.

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java (hunk 2 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: it reflects the xml node name, it has groovy 2 times.

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java (hunk 3 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: Done

File: src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java (hunk 4 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/eiffelcommons/helpers/JenkinsXmlData.java]: Done

**Pull request merged**